### DISCUSSION TITLE
Hold Charge - can it be managed?

### DISCUSSION THREAD

**Discussion by Yuffi17** [springfall2008/batpred#1328] [Q&A]
How do I set up Predbat so it doesn't use "Hold Charge" when solar input is available? 

I have a PV and battery system and want to  buy as little electricity from the grid as possible. As I understand it, "Hold Charge" exists to maintain battery levels ahead of anticipated use. However I am finding that it seems to occur when it's not needed, and can have the consequence of drawing from the grid when battery capacity is available and should be used.

For example, I'm currently generating 1.8kW solar, and the house load is 350W. Predbat is "holding charge" which sort of makes sense as the solar input exceeds the house load, but Hold Charge means any excess load will be taken from the grid and not from the battery (which is not good when the battery is fully charged and will probably top up afterwards on solar anyway).  

Trefor kindly added the Self-Sufficiency metric to help reduce grid import, but I am still seeing Hold Charge occur even with a self_sufficiency of 10p.  

Any help understanding why this is happening, and how to stop it, is very welcome.

**Comment by cyberkryten** [comment]
Are you charging a car, or do you have one plugged in on Octopus Intelligent Go?  I've seen my system go into 'Hold' mode when that kicks in.  If you post your plan and the debug output it may be obvious to someone why it has gone into Hold.

**Reply by Yuffi17** [reply]
No car and I'm on Agile for import tariff. 

Here's an example of where I don't think I need "Hold Charge" because the import price is higher than the export (so it doesn't make financial sense) and the solar is expected to provide enough for the house. I have two 9.5kWh batteries and use about 13kWh a day, so there is no need to hold charge for use during the evening (and any charge taken from the battery in the hold charge period is likely to be replaced via solar charging today so we'll start the evening with the batteries at 100% anyway).

![image](https://github.com/user-attachments/assets/0214fbfb-dddf-41fb-8b24-fa98821b35c1)

**Comment by jimbaxter** [comment]
I would also like to be able to control the hold power as I have a subtly different problem related to this issue.

With the Solax modbus API if the battery is at 100% and Solar is available then the solar panel's power is reduced to nearly 0W if any charge control is enabled. 
This is mainly an issue when the "hold charge" is used as discharging resolves the issue and charging rarely happens in this situation.

I could fix it with an Automation trigger that disables the Solax modbus battery control when the battery reaches 100% and predbat is in an active mode but I guess that may cause confusion in predbatt.

**Reply by gcoan** [reply]
Hi Jim

Suggest you raise this as a separate issue with the Solax modbus API in Predbat.  Might be able to change the service calls so that predbat doesn't stop your PV.

In general Predbat will optimise for cost and won't charge the battery unnecessarily - hence the use of hold charge and freeze charge.  Increasing metric self sufficiency should encourage Predbat to use solar rather than import it
